# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/TransportSecurityInfo.h
# Commit: d3482abe1fc9
# Full Hash: d3482abe1fc9117fa433abcf693558ad4ee2bb40
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2019-10-18 21:48:04
# Regressor Bug: 1580315
# File Overlap Count: 2
# Description:
#   Bug 1580315 - Use cert array to do certList serialization r=keeler
#   
#   The internal representation of certList has been converted to
#   cert array, and this patch does it for the serialization.
#   
# ==============================================================================

diff -r 6acfae11770c -r d3482abe1fc9 security/manager/ssl/TransportSecurityInfo.h
--- a/security/manager/ssl/TransportSecurityInfo.h	Fri Oct 18 15:31:49 2019 +0000
+++ b/security/manager/ssl/TransportSecurityInfo.h	Fri Oct 18 13:42:54 2019 +0000
@@ -145,6 +145,14 @@
   static nsresult ConvertCertListToCertArray(
       const nsCOMPtr<nsIX509CertList>& aCertList,
       nsTArray<RefPtr<nsIX509Cert>>& aCertArray);
+
+  // This function is used to read the binary that are serialized
+  // by using nsIX509CertList
+  nsresult ReadCertList(nsIObjectInputStream* aStream,
+                        nsTArray<RefPtr<nsIX509Cert>>& aCertList);
+  nsresult ReadCertificatesFromStream(nsIObjectInputStream* aStream,
+                                      uint32_t aSize,
+                                      nsTArray<RefPtr<nsIX509Cert>>& aCertList);
 };
 
 class RememberCertErrorsTable {